lib/bolt/config/transport/base.rb

```diff
@@ -77,6 +77,7 @@ def resolve(plugins)
           @resolved = true
 
           validate
+          self
```

Comment:
Is this not a bug in how we are resolving? https://github.com/puppetlabs/bolt/blob/e3a7529dc289db69fa39db6cd629ab8a23f25d2a/lib/bolt/inventory/inventory.rb#L74 It seems like maybe instead of `transform_values` that should just be `each_value` and the `resolve` will mutate the actual instance of the transport config?

Reply:
Something like this:
```diff
@@ -77,7 +77,6 @@ module Bolt
           @resolved = true
 
           validate
-          self
         end
 
         def resolved?
diff --git a/lib/bolt/inventory/inventory.rb b/lib/bolt/inventory/inventory.rb
index e3327231..bccad996 100644
--- a/lib/bolt/inventory/inventory.rb
+++ b/lib/bolt/inventory/inventory.rb
@@ -71,7 +71,8 @@ module Bolt
           @config
         else
           @config_resolved = true
-          @config.transform_values! { |t| t.resolved? ? t : t.resolve(@plugins) }
+          @config.each_value { |t| t.resolved? ? t : t.resolve(@plugins) }
+          @config
         end
       end
 
```

Reply:
Ok, I *tried* this first too, and was still seeing the same issue - which seemed weird, but the other solution worked. Of course now it's working, I must have just been doing something else wrong :P